src/main/java/me/jellysquid/mods/sodium/client/gui/SodiumGameOptionPages.java

```diff
@@ -27,11 +27,20 @@ public static OptionPage general() {
         List<OptionGroup> groups = new ArrayList<>();
 
         groups.add(OptionGroup.createBuilder()
+                .add(OptionImpl.createBuilder(int.class, vanillaOpts)
+                        .setName("GUI Scale")
+                        .setTooltip("Controls how many pixels wide/high a pixel of the GUI is, effectively scaling it. Set to 0 for automatic scaling.")
+                        .setControl(option -> new SliderControl(option, 0, 16, 1, v -> v == 0 ? "Automatic" : v + "x"))
```

Comment:
Setting the max to 16 by default will probably confuse users as with most monitors it doesn't reach that high. Ideally this would check the dimensions of the screen and then reduce the number accordingly, but this causes issues when resizing the window while the sodium gui is open. MC provides an automatic way to do this, through `MinecraftClient.getInstance().getWindow().calculateScaleFactor(0, MinecraftClient.getInstance().forcesUnicodeFont())`, but listening to the resize is the issue here.

Reply:
Screen has a method called `resize`, I think vanilla uses it for adjusting as well.

Reply:
I'm not sure that's better. Perhaps a lower maximum like 8 would do, but I wouldn't want it to have a lower maximum just because my window is smaller at the time i'm looking at it.

Reply:
Maybe change the name to something like "Max GUI Scale" and in the tooltip it shows what the current possible maximum is. (A greyed out, not interactible slider bar to also show this would be awesome, but probably too complex to be worth it)

So the maximum doesn't go down just because the current screen size doesn't support it and it tells users what the actual scale would be.